src/vw/src/main/scala/VowpalWabbitClassifier.scala

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import org.apache.spark.ml.param._
+import org.apache.spark.ml.util._
+import org.apache.spark.ml.classification.{ProbabilisticClassificationModel, ProbabilisticClassifier}
+import org.apache.spark.ml.linalg.{Vector, Vectors}
+import org.apache.spark.ml.param.shared.HasProbabilityCol
+import org.apache.spark.sql._
+import org.apache.spark.sql.functions.{col, udf}
+
+import scala.reflect.runtime.universe.{TypeTag, typeTag}
+import scala.math.exp
+
+object VowpalWabbitClassifier extends DefaultParamsReadable[VowpalWabbitClassifier]
+
+@InternalWrapper
+class VowpalWabbitClassifier(override val uid: String)
+  extends ProbabilisticClassifier[Row, VowpalWabbitClassifier, VowpalWabbitClassificationModel]
+  with VowpalWabbitBase
+{
+  def this() = this(Identifiable.randomUID("VowpalWabbitClassifier"))
+
+  override protected def train(dataset: Dataset[_]): VowpalWabbitClassificationModel = {
+
+    val binaryModel = trainInternal(dataset)
+
+    new VowpalWabbitClassificationModel(uid, binaryModel, getLabelCol, getFeaturesCol, getAdditionalFeatures,
+      getPredictionCol, getProbabilityCol, getRawPredictionCol)
+  }
+
+  override def copy(extra: ParamMap): VowpalWabbitClassifier = defaultCopy(extra)
+}
+
+// Preparation for multi-class learning, though it no fun as numClasses is spread around multiple reductions
+@InternalWrapper
+class VowpalWabbitClassificationModel(
+    override val uid: String, val model: Array[Byte], labelColName: String,
+    featuresColName: String, additionalFeaturesName: Array[String],
+    predictionColName: String, probabilityColName: String, rawPredictionColName: String)
+  extends ProbabilisticClassificationModel[Row, VowpalWabbitClassificationModel]
+    with VowpalWabbitBaseModel with HasProbabilityCol // TODO: HasThresholds
```

Comment:
you shouldn't need HasProbabilityCol as you get it already from ProbabilisticClassificationModel?

Reply:
same thing for HasThresholds

Reply:
done